src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSScoreUtils.java

```diff
@@ -0,0 +1,426 @@
+                getOrAddScoreInfo(taxId, taxIdsToScores, tree).reads += hit.numMates;
+            }
+        }
+
+        //Scores normalized to 100%
+        for (final Map.Entry<String, PSPathogenTaxonScore> entry : taxIdsToScores.entrySet()) {
+            final String readName = entry.getKey();
+            final PSPathogenTaxonScore score = entry.getValue();
+            if (sum == 0) {
+                score.scoreNormalized = 0;
+            } else {
+                score.scoreNormalized = 100.0 * score.score / sum;
+            }
+            taxIdsToScores.replace(readName, score);
+        }
+        PSUtils.logItemizedWarning(logger, invalidIds, "The following taxonomic ID hits were ignored because " +
+                "they either could not be found in the tree or had a reference length of 0 (this may happen when " +
+                "the catalog file, taxdump file, and/or pathogen reference are inconsistent)");
+        return taxIdsToScores;
+    }
+
+    /**
+     * Helper function for handling PSPathogenTaxonScore retrieval from the taxScores map
+     */
+    private static PSPathogenTaxonScore getOrAddScoreInfo(final String taxIds,
+                                                          final Map<String, PSPathogenTaxonScore> taxScores,
+                                                          final PSTree tree) {
+        final PSPathogenTaxonScore score;
+        if (taxScores.containsKey(taxIds)) {
+            score = taxScores.get(taxIds);
+        } else {
+            score = new PSPathogenTaxonScore();
+            score.refLength = tree.getLengthOf(taxIds);
+            taxScores.put(taxIds, score);
+        }
+        return score;
+    }
+
+    /**
+     * Reads taxonomy database that has been written using PSUtils.writeKryoTwo()
+     */
+    @SuppressWarnings("unchecked")
+    public static PSTaxonomyDatabase readTaxonomyDatabase(final String filePath) {
+        final Kryo kryo = new Kryo();
+        kryo.setReferences(false);
+        final Input input = new Input(BucketUtils.openFile(filePath));
+        final PSTaxonomyDatabase taxonomyDatabase = kryo.readObject(input, PSTaxonomyDatabase.class);
+        input.close();
+        return taxonomyDatabase;
+    }
+
+    /**
+     * Output a tab-delimited table of taxonomic scores
+     */
+    public static void writeScoresFile(final Map<String, PSPathogenTaxonScore> scores,
+                                       final PSTree tree, final String filePath) {
+        try {
+            final OutputStream file = BucketUtils.createFile(filePath);
+            final String header = "tax_id\ttaxonomy\ttype\tname\t" + PSPathogenTaxonScore.outputHeader + "\n";
+            file.write(header.getBytes(Charset.defaultCharset()));
+            for (final String key : scores.keySet()) {
+                final String name = tree.getNameOf(key).replace(" ", "_");
+                final String rank = tree.getRankOf(key).replace(" ", "_");
```

Comment:
Probably not but I've combined all the replace()'s into just one at the end now.